apps/rehash.c

```diff
@@ -553,6 +553,9 @@ int rehash_main(int argc, char **argv)
     evpmd = EVP_sha1();
     evpmdsize = EVP_MD_get_size(evpmd);
 
+    if (evpmdsize <= 0)
```

Comment:
I don't think it ever happens currently, but it might in the future occur that EVP_MD_get_size returns a value greater than EVP_MAX_MD_SIZE, which is the length the ep->digest array is defined to. You may want to consider adding an additional check for that here, to prevent weird memory overruns in the future.

Reply:
Thanks. I have submitted another commit to add an addtional check.